src/engine/docker_engine.rs

```diff
@@ -228,3 +250,29 @@ impl Engine for DockerEngine {
         }
     }
 }
+
+/// Returns a map of all traefik related labels, used for networking purposes
+/// Note that it always redirects to port 80
+fn build_traefik_labels(app: &Application) -> HashMap<String, String> {
+    return HashMap::from([
+        ("traefik.enable".into(), "true".into()),
+        (
+            format!("traefik.http.routers.{}.entrypoints", &app.application_id),
+            "websecure".into(),
+        ),
+        (
+            format!(
+                "traefik.http.services.{}.loadbalancer.server.port",
+                &app.application_id
+            ),
+            "80".into(),
+        ),
+        (
+            format!("traefik.http.routers.{}.rule", &app.application_id),
+            format!(
+                "Host(`{}.user-app.{}`)",
+                &app.application_id, TRAEFIK_CONFIG.fqdn
+            ),
+        ),
+    ]);
```

Comment:
May remove unnecessary keyword `return` here (and final `;` if removed) ; may remove `&` from `&app`, as given app is already borrowed (see precedent comment)
```suggestion
    HashMap::from([
        ("traefik.enable".into(), "true".into()),
        (
            format!("traefik.http.routers.{}.entrypoints", app.application_id),
            "websecure".into(),
        ),
        (
            format!(
                "traefik.http.services.{}.loadbalancer.server.port",
                app.application_id
            ),
            "80".into(),
        ),
        (
            format!("traefik.http.routers.{}.rule", app.application_id),
            format!(
                "Host(`{}.user-app.{}`)",
                app.application_id, TRAEFIK_CONFIG.fqdn
            ),
        ),
    ])
```